std/uni.d

```diff
@@ -8771,20 +8777,27 @@ else
 // trusted -> avoid bounds check
 @trusted pure nothrow @nogc private
 {
-    import std.internal.unicode_tables; // : toLowerTable, toTitleTable, toUpperTable; // generated file
-
     // hide template instances behind functions (Bugzilla 13232)
     ushort toLowerIndex(dchar c) { return toLowerIndexTrie[c]; }
     ushort toLowerSimpleIndex(dchar c) { return toLowerSimpleIndexTrie[c]; }
-    dchar toLowerTab(size_t idx) { return toLowerTable[idx]; }
+    dchar toLowerTab()(size_t idx) {
+        import std.internal.unicode_tables : toLowerTable; // generated file
+        return toLowerTable[idx];
+    }
 
     ushort toTitleIndex(dchar c) { return toTitleIndexTrie[c]; }
     ushort toTitleSimpleIndex(dchar c) { return toTitleSimpleIndexTrie[c]; }
-    dchar toTitleTab(size_t idx) { return toTitleTable[idx]; }
+    dchar toTitleTab()(size_t idx) {
+        import std.internal.unicode_tables : toTitleTable; // generated file
+        return toTitleTable[idx];
+    }
 
     ushort toUpperIndex(dchar c) { return toUpperIndexTrie[c]; }
     ushort toUpperSimpleIndex(dchar c) { return toUpperSimpleIndexTrie[c]; }
-    dchar toUpperTab(size_t idx) { return toUpperTable[idx]; }
+    dchar toUpperTab()(size_t idx) {
```

Comment:
Why not a function?

Actually making things a function above was a decision to also save on codegen time when sth. gets used, because the binary is already in phobos. Imports in function bodies do not leak to the call site, so you also save those when merely calling.
The compiler can still inline things if found appropriate at which point you pay the same price as for instantiating the template.